src/stores/ResourceConfig.tsx

```diff
@@ -194,16 +237,134 @@ const getInitialState = (
+
     setCurrentCollection: (value) => {
         set(
             produce((state: ResourceConfigState) => {
-                state.currentCollection = value;
+                const {
+                    collections,
+                    collectionRemovalMetadata: {
+                        selectedCollection,
+                        removedCollection,
+                        index: removedCollectionIndex,
+                    },
+                } = get();
+
+                const collectionCount = collections?.length;
+
+                if (value && collections?.includes(value)) {
+                    state.currentCollection = value;
+                } else if (
+                    collectionCount &&
+                    selectedCollection === removedCollection
+                ) {
+                    if (
+                        removedCollectionIndex > -1 &&
+                        removedCollectionIndex < collections.length
+                    ) {
+                        state.currentCollection =
+                            collections[removedCollectionIndex];
+                    } else if (removedCollectionIndex === collections.length) {
+                        state.currentCollection =
+                            collections[removedCollectionIndex - 1];
+                    }
+                } else if (collectionCount && removedCollection === value) {
+                    state.currentCollection = selectedCollection;
+                } else {
+                    state.currentCollection = null;
+                }
             }),
             false,
             'Current Collection Changed'
         );
     },
 
+    setDiscoveredCollections: (value) => {
+        set(
+            produce((state: ResourceConfigState) => {
+                const discoveredCollections: string[] = [];
+
+                value.spec.bindings.forEach((binding: any) => {
+                    discoveredCollections.push(binding.target);
+                });
+
+                state.discoveredCollections = discoveredCollections;
+            }),
+            false,
+            'Discovered Collections Set'
+        );
+    },
+
+    setRestrictedDiscoveredCollections: (value) => {
+        set(
+            produce((state: ResourceConfigState) => {
+                const {
+                    discoveredCollections,
+                    restrictedDiscoveredCollections,
+                } = get();
+
+                if (restrictedDiscoveredCollections.includes(value)) {
+                    state.restrictedDiscoveredCollections =
+                        restrictedDiscoveredCollections.filter(
+                            (collection) => collection !== value
+                        );
+                } else if (discoveredCollections?.includes(value)) {
+                    state.restrictedDiscoveredCollections = [
+                        value,
+                        ...restrictedDiscoveredCollections,
+                    ];
+                }
```

Comment:
I think this might be a spot where it makes sense to do the setting of `state.restrictedDiscoveredCollections` on a single line and storing what we're gonna put there in a variable.

Unless you think eventually there will be more logic difference in the two `if` blocks.

Reply:
If that is what you feel is best.